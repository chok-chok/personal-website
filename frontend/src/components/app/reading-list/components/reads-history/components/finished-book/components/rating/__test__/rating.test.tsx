import * as React from 'react';
import Rating from '../rating';
import { shallow } from 'enzyme';

describe('Test Rating component', () => {
  it('should renders correctly', () => {
    const tree = shallow(<Rating rating={5} />);

    expect(tree).toMatchSnapshot();
  });

  it('should renders correct number of stars base on prop', () => {
    const oneStartRatingWrapper = shallow(<Rating rating={1} />);
    const twoStartRatingWrapper = shallow(<Rating rating={2} />);
    const threeStartRatingWrapper = shallow(<Rating rating={3} />);
    const fourStartRatingWrapper = shallow(<Rating rating={4} />);
    const fiveStarRatingWrapper = shallow(<Rating rating={5} />);

    expect(oneStartRatingWrapper.find('.black-rate-star')).toHaveLength(1);
    expect(oneStartRatingWrapper.find('.white-rate-star')).toHaveLength(4);

    expect(twoStartRatingWrapper.find('.black-rate-star')).toHaveLength(2);
    expect(twoStartRatingWrapper.find('.white-rate-star')).toHaveLength(3);

    expect(threeStartRatingWrapper.find('.black-rate-star')).toHaveLength(3);
    expect(threeStartRatingWrapper.find('.white-rate-star')).toHaveLength(2);

    expect(fourStartRatingWrapper.find('.black-rate-star')).toHaveLength(4);
    expect(fourStartRatingWrapper.find('.white-rate-star')).toHaveLength(1);

    expect(fiveStarRatingWrapper.find('.black-rate-star')).toHaveLength(5);
    expect(fiveStarRatingWrapper.find('.white-rate-star')).toHaveLength(0);
  });
});
