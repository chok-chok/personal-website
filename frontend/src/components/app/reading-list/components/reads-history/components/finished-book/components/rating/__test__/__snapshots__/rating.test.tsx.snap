// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rating component should renders correctly 1`] = `
<Fragment>
  <styled__RatingWrapper>
    <styled__RatingStartImg
      className="black-rate-star"
      key="1"
      src="/svg/star-black.svg"
    />
    <styled__RatingStartImg
      className="black-rate-star"
      key="2"
      src="/svg/star-black.svg"
    />
    <styled__RatingStartImg
      className="black-rate-star"
      key="3"
      src="/svg/star-black.svg"
    />
    <styled__RatingStartImg
      className="black-rate-star"
      key="4"
      src="/svg/star-black.svg"
    />
    <styled__RatingStartImg
      className="black-rate-star"
      key="5"
      src="/svg/star-black.svg"
    />
  </styled__RatingWrapper>
</Fragment>
`;
